shared-data/python/opentrons_shared_data/protocol/models/protocol_schema_v6.py

```diff
@@ -38,7 +38,7 @@ class Params(BaseModel):
     pipetteId: Optional[str]
     mount: Optional[str]
     moduleId: Optional[str]
-    location: Optional[Location]
+    location: Union[Optional[Location], Literal['offDeck']]
```

Comment:
I'd make this an Optional of Union rather than Union of Optional. Even though I think the effect is the same, we want to be sure in conveying that this location params is not required for all commands.

```suggestion
    location: Optional[Union[Location, Literal["offDeck"]]]
```